cmd/updater/main.go

```diff
@@ -402,6 +404,111 @@ func AppendColumn(headers []string, format nameConfig, grid *state.Grid, rows ma
 	}
 }
 
+// alertRows configures the alert for every row that has one.
+func alertRows(cols []*state.Column, rows []*state.Row, openFailures, closePasses int) {
+	for _, r := range rows {
+		r.AlertInfo = alertRow(cols, r, openFailures, closePasses)
+	}
+}
+
+// alertRow returns an AlertInfo proto if there have been failuresToOpen consecutive failures more recently than passesToClose.
+func alertRow(cols []*state.Column, row *state.Row, failuresToOpen, passesToClose int) *state.AlertInfo {
+	if failuresToOpen == 0 {
+		return nil
+	}
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	var failures int
+	var totalFailures int32
+	var passes int
+	var compressedIdx int
+	ch := result.Iter(ctx, row.Results)
+	var lastFail *state.Column
+	var latestPass *state.Column
+	var failIdx int
+	for _, col := range cols {
```

Comment:
nit: Function looks good! Some extra comments would be great.

Reply:
Done